hooks/useClickOutside.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (
+  // eslint-disable-next-line no-unused-vars
+  event: MouseEvent | TouchEvent
+) => void;
+
+export const useClickOutside = <T extends HTMLElement>(
+  handler: useClickOutsideType
+) => {
+  const ref = useRef<T>();
```

Comment:
You need to pass `null` to `useRef` for it to infer the correct type, that's why the build is failing. 
```suggestion
  const ref = useRef<T>(null);
```